VocaDbModel/Service/VideoServices/VideoServiceBilibili.cs

```diff
@@ -53,18 +54,19 @@ public VideoServiceBilibili()
 		public override async Task<VideoUrlParseResult> ParseByUrlAsync(string url, bool getTitle) {
 
 			var id = GetIdByUrl(url);
+			var aid = id.StartsWith("BV") ? BilibiliHelper.Decode(id).ToString() : id;
 
-			if (string.IsNullOrEmpty(id))
+			if (string.IsNullOrEmpty(aid))
 				return VideoUrlParseResult.CreateError(url, VideoUrlParseResultType.NoMatcher, "No matcher");
 
 			if (!getTitle) {
-				return VideoUrlParseResult.CreateOk(url, PVService.Bilibili, id, VideoTitleParseResult.Empty);
+				return VideoUrlParseResult.CreateOk(url, PVService.Bilibili, aid, VideoTitleParseResult.Empty);
 			}
 
-			var paramStr = string.Format("appkey={0}&id={1}&type=json{2}", AppConfig.BilibiliAppKey, id, AppConfig.BilibiliSecretKey);
+			var paramStr = string.Format("appkey={0}&id={1}&type=json{2}", AppConfig.BilibiliAppKey, aid, AppConfig.BilibiliSecretKey);
 			var paramStrMd5 = CryptoHelper.HashString(paramStr, CryptoHelper.MD5).ToLowerInvariant();
 
-			var requestUrl = string.Format("https://api.bilibili.com/view?appkey={0}&id={1}&type=json&sign={2}", AppConfig.BilibiliAppKey, id, paramStrMd5);
+			var requestUrl = string.Format("https://api.bilibili.com/view?appkey={0}&id={1}&type=json&sign={2}", AppConfig.BilibiliAppKey, aid, paramStrMd5);

```

Comment:
Actually, we should replace this to use the new API, as discussed in #581. Apparently the new API doesn't even need an API key, so you can test it properly, and no need to do the hacky encoding/decoding either.